fg/apps/factory/views/twist.py

```diff
@@ -369,7 +369,7 @@ def import_plate_task(rows, fields, factory_order):
 
                    # Look for a sample already associated with plate from the factory order      
                     for plate_well in plate.wells.all():
-                        contenders = Sample.objects.filter(wells=plate_well)
+                        contenders = Sample.objects.filter(wells=plate_well).filter(part=part)
                         if contenders.count() > 0:
```

Comment:
Are you sure that here you don't want:

```python
contenders = Sample.objects.filter(wells__contains=plate_well).filter(part=part)
```
since wells is a set?

Reply:
I don't know, do I want that? The code worked in my hands, not sure what `wells__contains` would improve that. Is there some advantage I'd be looking at?

Reply:
Just tested, we do want equals here and not contains.

Reply:
Ok, well then this pull is done

Reply:
Yes I'm aware, I haven't been able to merge because I'm responding to all those other issues that someone is opening :)